src/js/AccessibilityUtils.js

```diff
@@ -577,6 +577,44 @@ axs.utils.hasLabel = function(element) {
 };
 
 /**
+ * Determine if this element natively supports being disabled (i.e. via the `disabled` attribute.
+ * Disabled here means that the element should be considered disabled according to specification.
+ * This element may or may not be effectively disabled in practice as this is dependent on implementation.
+ *
+ * @param {Element} element An element to check.
+ * @return {boolean} true If the element supports being natively disabled.
+ */
+axs.utils.isNativelyDisableable = function(element) {
+    var tagName = element.tagName.toUpperCase();
+    return (tagName in axs.constants.NATIVELY_DISABLEABLE);
+};
+
+/**
+ * Determine if this element is disabled directly or indirectly by a disabled ancestor.
+ * Disabled here means that the element should be considered disabled according to specification.
+ * This element may or may not be effectively disabled in practice as this is dependent on implementation.
+ *
+ * @param {Element} element An element to check.
+ * @return {boolean} true if the element or one of its ancestors is disabled.
+ */
+axs.utils.isElementDisabled = function(element) {
+    if (axs.browserUtils.matchSelector(element, '[aria-disabled=true], [aria-disabled=true] *')) {
+        return true;
+    }
+    if(!axs.utils.isNativelyDisableable(element) ||
+            axs.browserUtils.matchSelector(element, 'fieldset>legend:first-of-type *')) {
+        return false;
+    }
+    var next = element;
+    do {
```

Comment:
Love it!
I have no excuse for not coming up with that myself :( 

It's not a pattern I have got into the habit of using but it works so elegantly here I'm going to have to review all the JS I have written at work and look for places where I should have done this.